pkg/sql/sem/builtins/geo_builtins.go

```diff
@@ -7898,3 +7882,16 @@ func geosVersion() string {
 	}
 	return geosV
 }
+
+func asMVTGeometry(
+	g geo.Geometry, bounds geo.CartesianBoundingBox, extent int, buffer int, clipGeometry bool,
+) (tree.Datum, error) {
+	newGeom, err := geomfn.AsMVTGeometry(g, bounds, extent, buffer, clipGeometry)
+	if err != nil {
+		return nil, err
+	}
+	if newGeom == nil {
```

Comment:
if you wanted to micro optimise, you could leave the return of AsMVTGeometry as an empty geometry struct instead of making it a pointer and do `if newGeom ==  geo.Geometry {` (there's prior art doing this iirc and it makes a sizable difference)